tests/alt-registry.rs

```diff
@@ -211,9 +212,14 @@ fn cannot_publish_with_registry_dependency() {
 
     Package::new("bar", "0.0.1").alternative(true).publish();
 
+    // Login so that we have the token available
+    assert_that(p.cargo("login").masquerade_as_nightly_cargo()
+                 .arg("--registry").arg("alternative").arg("TOKEN").arg("-Zunstable-options"),
+                execs().with_status(0));
+
     assert_that(p.cargo("publish").masquerade_as_nightly_cargo()
-                 .arg("--index").arg(registry::alt_registry().to_string()),
-                execs().with_status(101));
```

Comment:
Could this perhaps stick around as a separate test? I think we'll want an assertion that you can't publish a crate with dependencies on a separate registry to crates.io

Reply:
I have added the test back.